packages/env/lib/download-sources.js

```diff
@@ -99,69 +99,34 @@ async function downloadSource( source, options ) {
 async function downloadGitSource( source, { onProgress, spinner, debug } ) {
 	const log = debug
 		? ( message ) => {
-				spinner.info( `NodeGit: ${ message }` );
+				spinner.info( `SimpleGit: ${ message }` );
 				spinner.start();
 		  }
 		: () => {};
 	onProgress( 0 );
 
-	const gitFetchOptions = {
-		fetchOpts: {
-			callbacks: {
-				transferProgress( progress ) {
-					// Fetches are finished when all objects are received and indexed,
-					// so received objects plus indexed objects should equal twice
-					// the total number of objects when done.
-					onProgress(
-						( progress.receivedObjects() +
-							progress.indexedObjects() ) /
-							( progress.totalObjects() * 2 )
-					);
-				},
-				certificateCheck: () => 0,
-				credentials: ( url, userName ) => {
-					try {
-						return NodeGit.Cred.sshKeyFromAgent( userName );
-					} catch {
-						return NodeGit.Cred.defaultNew();
-					}
-				},
-			},
-		},
-	};
-
 	log( 'Cloning or getting the repo.' );
-	const repository = await NodeGit.Clone(
-		source.url,
-		source.clonePath,
-		gitFetchOptions
-	).catch( () => {
-		log( 'Repo already exists, get it.' );
-		return NodeGit.Repository.open( source.clonePath );
-	} );
+	const git = SimpleGit();
+
+	const isRepo =
+		fs.existsSync( source.clonePath ) &&
+		( await git.cwd( source.clonePath ).checkIsRepo() );
+
+	if ( isRepo ) {
+		log( 'Repo already exists, using it.' );
+	} else {
+		await git.clone( source.url, source.clonePath, {
+			'--depth': '1',
```

Comment:
One thing I was curious about when I started putting this together: with a shallow clone, will git.checkout still work for refs which are older commits or tags?

Reply:
That's a great question!

Only one way to find out :wink:

Reply:
I just tried with

```diff
diff --git a/.wp-env.json b/.wp-env.json
index ccd1231aa2..4d67d71c82 100644
--- a/.wp-env.json
+++ b/.wp-env.json
@@ -3,7 +3,7 @@
        "plugins": [
                "."
        ],
-       "themes": [ "WordPress/theme-experiments/tt1-blocks" ],
+       "themes": [ "WordPress/theme-experiments/tt1-blocks#tt1-blocks@0.4.3" ],
        "env": {
                "tests": {
                        "mappings": {
```

(see https://github.com/WordPress/gutenberg/pull/28741#issuecomment-775320560), and it worked :tada: